### PR TITLE
Feature/UI theme

### PR DESCRIPTION
**Pull request by pujiaxin33**
## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Check list
1. [ ] Does your submission pass tests?
2. [x] Have you lint your code locally prior to submission?
3. [ ] Have you update the wiki of your change?

## Reference
- Card:  https://trello.com/c/kxxzQkkd/58-3-ui-themes-1
- Related PR: The related pr of current one.

### CODE REVIEW

File: modules/UI/Sources/Dimension/Dimensions.swift [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import Foundation
+import UIKit
+
+public extension CGFloat {
+    // spacing
+    static let standardSpacing: CGFloat = 20.0
+    static let smallSpacing: CGFloat = 4.0
+    static let halfSpacing: CGFloat = 10.0
+    // dimension
+    static let standardDimension5 = 5.0
```
Comment (Tony---Zhang) [inline modules/UI/Sources/Dimension/Dimensions.swift]: 直接叫”dimension5“怎么样？
我们不提供这种单数的dimension，只提供4的倍数的dimension，单数的提供一个dimension1跟dimension2来处理divider啥的情况。

File: modules/UI/Sources/Theme/Colors/ThemeColors.swift [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    case onSurface
+    case onError
+    case statusBarColor
+}
+
+public extension ThemeColors {
+    var color: Color {
+        switch self {
+        case .statusBarColor:
+            return ColorName.blue.color
```
Comment (Tony---Zhang) [inline modules/UI/Sources/Theme/Colors/ThemeColors.swift]: 这里不需要单独对statusBar做特殊操作吧？

File: modules/UI/Sources/Theme/Colors/ColorName.swift [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    case gray400
+    case gray500
+    case gray600
+    case gray700
+    case gray800
+    case gray900
+    // blue
+    case blue
+    // light blue
+    case lightBlue50
```
Comment (Tony---Zhang) [inline modules/UI/Sources/Theme/Colors/ColorName.swift]: We prefer `blue50`

File: modules/UI/Sources/Theme/Colors/ThemeColors.swift (hunk 2 above)
Comment (pujiaxin33) [reply modules/UI/Sources/Theme/Colors/ThemeColors.swift]: 我的想法是Color枚举相当于颜色值仓库，其他所有使用颜色值的地方都可以从这里取值。然后ThemeColors，是基于UI Guide设计的一个应用主题色集合，它可以自定义颜色值，也可以使用Color里面定义好的色值。

File: modules/UI/Sources/Dimension/Dimensions.swift (hunk 1 above)
Comment (pujiaxin33) [reply modules/UI/Sources/Dimension/Dimensions.swift]: 可以

File: modules/UI/Sources/Theme/Colors/ColorName.swift (hunk 3 above)
Comment (pujiaxin33) [reply modules/UI/Sources/Theme/Colors/ColorName.swift]: 参考的是material guide

File: modules/UI/Sources/Theme/Colors/ThemeColors.swift (hunk 2 above)
Comment (Tony---Zhang) [reply modules/UI/Sources/Theme/Colors/ThemeColors.swift]: 那写一个函数或者注释说一下这个意图？

File: modules/UI/Sources/Dimension/Dimensions.swift (hunk 1 above)
Comment (Tony---Zhang) [reply modules/UI/Sources/Dimension/Dimensions.swift]: object Dimensions {
    val dimension1 = 1.dp
    val dimension2 = 2.dp
    val dimension4 = 4.dp
    val dimension8 = 8.dp
    val dimension12 = 12.dp
    val dimension16 = 16.dp
    val dimension20 = 20.dp
    val dimension24 = 24.dp
    val dimension28 = 28.dp
    val dimension32 = 32.dp
    val dimension36 = 36.dp
    val dimension40 = 40.dp
    val dimension44 = 44.dp
    val dimension48 = 48.dp

    val standardPadding = dimension16
    val halfPadding = dimension8
    val smallPadding = dimension4

    val divider = dimension1
    val standardSpacing = dimension20

    val iconSmall = dimension24
    var iconMedium = dimension32
    var iconNormal = dimension48
}
参考这个吧，我准备把Android也改成这样（ios的icon默认可以改成44dp）

File: modules/UI/Sources/Theme/Colors/ColorName.swift [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+//  ColorName.swift
+//  UI
+//
+//  Copyright 2022 Thoughtworks, Inc. All rights reserved.
+//
+
+import Foundation
+import SwiftUI
+
+public enum ColorName: String {
```
Comment (Tony---Zhang) [inline modules/UI/Sources/Theme/Colors/ColorName.swift]: How about `Colors`, follow `Dimensions`

Comment (Tony---Zhang) [reply modules/UI/Sources/Theme/Colors/ColorName.swift]: Or we can call it `ColorPalette`, I prefer `Colors`

File: modules/UI/Sources/Dimension/Dimensions.swift (hunk 1 above)
Comment (pujiaxin33) [reply modules/UI/Sources/Dimension/Dimensions.swift]: ok

File: modules/UI/Sources/Theme/Colors/ColorName.swift (hunk 4 above)
Comment (pujiaxin33) [reply modules/UI/Sources/Theme/Colors/ColorName.swift]: Ok! I would like to adopt `Colors`

File: modules/UI/Sources/Theme/Colors/ThemeColors.swift (hunk 2 above)
Comment (pujiaxin33) [reply modules/UI/Sources/Theme/Colors/ThemeColors.swift]: 添加了注释

**Pull request merged**